core/services/ardupilot_manager/ArduPilotManager.py

```diff
@@ -138,7 +138,7 @@ def add_new_endpoint(self, endpoint: Endpoint) -> bool:
             self.configuration.get("endpoints", []).remove(endpoint.asdict())
             return False
         self.settings.save(self.configuration)
-        return True
+        return self.restart()
```

Comment:
I think that is not a good solution, the API should request restart, otherwise if someone wants to add/remove multiple endpoints will end up with multiple resets of the tool. We should restart only once and when is really required.

Reply:
We can go with one of two paths:
1. Expose an endpoint (on the REST API) for manager restarting (meaning that the frontend will need to know when to call the restart).
2. Modifying the method on the manager to allow multiple endpoints to be added once (as it was before), an also it's corresponding method on the API (meaning the API would allow for multiple endpoints to be added once).

I like the second one more, as I think the frontend should not know how the internals of the manager work. Seems more reasonable.

Reply:
The front end should not know, but if the user changes a bunch of endpoints, adding and removing, a simple "apply" button will do the restart after doing all deletes/creations

Reply:
Yes! I was thinking about storing them on the frontend while being added and as soon as the user hits "apply", the API will be called with multiple endpoints at once. This makes everything very clear and we will have just one interface.

Reply:
I think that start/restart/stop routing is not something that needs to be exposed to the api/frontend. It is an implementation detail to be handled by the backend. We can eventually implement our own router that has no problem dynamically adding/removing endpoints and so a start/restart will not be required in the backend.

I think @rafaellehmkuhl suggestion to have the API to add multiple endpoints in a single request is the best solution to avoid unnecessary restarts of the backend tool.